Disabler SMG: We got laser carbine, so why not have automatic disabler?

DrDiasyl opened the pull request:
## About The Pull Request
Basically, it's #78685, but instead of an automatic deadly laser gun, it's an automatic disabler gun.

https://github.com/tgstation/tgstation/assets/42353186/6ce1c401-30a6-45f7-a55c-4d95f742c35a

_For comprasion, normal disabler takes 4 shoots and too uses 80% of charge_

![disabler SMG](https://github.com/tgstation/tgstation/assets/42353186/7242a0bd-9e78-4e98-a746-7c735b4f997e)

![smg inahnd](https://github.com/tgstation/tgstation/assets/42353186/a20848ab-9157-43a8-a146-6c3b57e0a817)

It's a side-grade as well, the damage is halved (15 instead of 30) while the ammo count (40 instead of 20) is higher. So damage potential stays the same.
The sound which it uses is more weak sounding to signify that.

You **can't** dual wield them.

You can order a crate of 3 at Cargo.
## Why It's Good For The Game
Automatic guns are fun, and as laser carbine showed - REAL FUN.
This introduces another automatic gun available to the crew through Cargo because variety is always good, and this way we cover the both most prominent projectiles (Lasers and Disablers). 
Now Security can have fun without only killing people, this also lets Pacificists have some as well.

<details>
  <summary>GET DISABLED</summary>

  ![HoS does ratatata](https://github.com/tgstation/tgstation/assets/42353186/29b0d1b3-3bae-4230-ac03-958c58d67a7e)

</details>

## Changelog
:cl:
add: New automatic weapon for the crew - Disabler SMG. Capable of rapidly firing weak disabler beams.
/:cl:

DrDiasyl committed:
Adds disabler smg

DrDiasyl committed:
inhands

DrDiasyl committed:
unique inhands

DrDiasyl committed:
ammo decrease

MrMelbert commented:
Using this on a riot looks fun as hell. 

The inhands look rather similar to normal disablers, I'd be worried about differentiating them at a glance.

xXPawnStarrXx commented:
Disablers > Lethals. Lethals are for cowards, you can always murder them after you stun them into stamcrit 😎

EricZilla commented:
As with thr Carbine I will suggest maybe making the lasers teal to signify they are different than disabler shots.
(My nefarious plan to make a lightshow is proceeding perfectly)

Wallemations reviewed (approved):
Clean sprite

Elli-Skala commented:
What if it was on a mech.

Jacquerel commented:
> What if it was on a mech.

I think those have a disabler shotgun instead

Cadunkus commented:
This seems too weak to compare with the regular disabler. Since it's a bulky item it can't be stored in backpacks unlike the regular disabler and most secoffs already have their armor and belt slot taken by other things (laser gun, security belt, shotgun, energy gun, etc.)

If it was normal sized like the disabler, then it would be a sidegrade. I'm thinking you should up the stamina damage to 20 from 15.

DrDiasyl commented:
> Using this on a riot looks fun as hell. 
> 
> The inhands look rather similar to normal disablers, I'd be worried about differentiating them at a glance. 

I have changed them to have longer barrel.

DrDiasyl commented:
> This seems too weak to compare with the regular disabler. Since it's a bulky item it can't be stored in backpacks unlike the regular disabler and most secoffs already have their armor and belt slot taken by other things (laser gun, security belt, shotgun, energy gun, etc.)
> 
> If it was normal sized like the disabler, then it would be a sidegrade. I'm thinking you should up the stamina damage to 20 from 15.

Hmm, you have a point about the size.
Will probably change it, as if this is the same power as disabler, the size should not really matter.

DrDiasyl committed:
Revert "ammo decrease"

This reverts commit 85d1c507db2a578b03be7846c36926aae2b9c3e5.

DrDiasyl committed:
Normal Sized now

This is a sidegreade so its the same power as normal disabler, so it should be smaller as well

SimplySmiley123 commented:
> Disablers > Lethals. Lethals are for cowards, you can always murder them after you stun them into stamcrit 😎

Real space_**MEN**_ fist fight to the death.

But anyway, fun idea, but with the size it's definitely not a sidegrade. Other then that only nitpick I got is the disabler bolts of it being the same as regular disabler's but, eh, you still know it's bad for your antagonistic life to be hit by it too many times.
_Also digging the sprite, looks pretty neat._

DrDiasyl commented:
> > Disablers > Lethals. Lethals are for cowards, you can always murder them after you stun them into stamcrit 😎
> 
> Real space**MEN** fist fight to the death.
> 
> But anyway, fun idea, but with the size it's definitely not a sidegrade. Other then that only nitpick I got is the disabler bolts of it being the same as regular disabler's but, eh, you still know it's bad for your antagonistic life to be hit by it too many times.
> _Also digging the sprite, looks pretty neat._

Size been changed.

And I don't really want to change laser color as disabler laser color is pale enough, making it more brighter will make it just white.

DrDiasyl committed:
Seclite mounting fix

ATH1909 commented:
any particular reason that the nonlethal version can't be dual-wielded?

Jacquerel commented:
> And I don't really want to change laser color as disabler laser color is pale enough, making it more brighter will make it just white.

How about making them smaller?

DrDiasyl commented:
> any particular reason that the nonlethal version can't be dual-wielded?

Same reason the laser carbine can't be dual-wielded - LASER HELL. (just balance)

Loafin34 commented:
honestly, it'd be cool if this was a modkit for the disabler instead of a standalone thing, as it looks very much like someone edited a disabler to hell and back.

deathrobotpunch commented:
finally my dumbass self can catch criminals with my shitty aim

Jacquerel commented on code/modules/projectiles/guns/energy/stun.dm:
```diff
@@ -40,6 +40,25 @@
+	icon_state = "disabler_smg"
+	ammo_type = list(/obj/item/ammo_casing/energy/disabler/smg)
+	shaded_charge = 1
+	var/allow_akimbo = FALSE
```
Any reason this is a var? It doesn't have any subtypes and an admin changing it at runtime won't do anything.

DrDiasyl replied:
> Any reason this is a var? It doesn't have any subtypes and an admin changing it at runtime won't do anything.

Laser carbine does the same, thought it makes sense.

GPeckman replied:
Hi, person who designed laser carbines here. It doesn't technically have to be a var, you could just pass `FALSE` to the component directly. I just thought making it explicit would be better.

Jacquerel commented on code/modules/projectiles/guns/energy/stun.dm:
```diff
@@ -40,6 +40,25 @@
+	AddComponent(/datum/component/automatic_fire, 0.15 SECONDS, allow_akimbo = allow_akimbo)
+
+/obj/item/gun/energy/disabler/add_seclight_point()
+	AddComponent(/datum/component/seclite_attachable, \
+		light_overlay_icon = 'icons/obj/weapons/guns/flashlights.dmi', \
+		light_overlay = "flight", \
+		overlay_x = 15, \
+		overlay_y = 13)
```
```suggestion
	AddComponent(\
		/datum/component/seclite_attachable, \
		light_overlay_icon = 'icons/obj/weapons/guns/flashlights.dmi', \
		light_overlay = "flight", \
		overlay_x = 15, \
		overlay_y = 13, \
	)
```

Jacquerel reviewed (changes requested):
I still think making the projectile sprite a little smaller is a good idea but these are the code reviews.

DrDiasyl committed:
Jacquerel requests

Co-Authored-By: Jacquerel <7483112+Jacquerel@users.noreply.github.com>

DrDiasyl commented:
Additional material:

https://youtu.be/WlV8KtpgBY4
